ndcube/ndcube.py

```diff
@@ -362,6 +362,93 @@ def axis_world_coords(self, *axes, edges=False):
         else:
             return tuple(axes_coords)
 
+    def axis_world_coords_values(self, *axes, edges=False):
+        """
+        Returns WCS coordinate values of all pixels for desired axes.
+
+        Parameters
+        ----------
+        axes: `int` or `str`, or multiple `int` or `str`
+            Axis number in numpy ordering or unique substring of
+            `~ndcube.NDCube.wcs.world_axis_physical_types`
+            of axes for which real world coordinates are desired.
+            axes=None implies all axes will be returned.
+
+        edges: `bool`
+            If True, the coords at the edges of the pixels are returned
+            rather than the coords at the center of the pixels.
+            Note that there are n+1 edges for n pixels which is reflected
+            in the returned coords.
+            Default=False, i.e. pixel centers are returned.
+
+        Returns
+        -------
+        axis_names: `tuple` of `str`
+            The physical types of the coords returned.
+
+        axes_coords: `tuple` of `astropy.units.Quantity`
+            Real world coords for axes requested by user.
+            Returned in same order as axis_names.
+
+        Example
+        -------
+        >>> NDCube.all_world_coords_values(('lat', 'lon')) # doctest: +SKIP
+        >>> NDCube.all_world_coords_values(2) # doctest: +SKIP
+
+        """
+        # Create meshgrid of all pixel coordinates.
+        # If user, wants edges, set pixel values to pixel edges.
+        # Else make pixel centers.
+        wcs_shape = self.data.shape[::-1]
+        if edges:
+            wcs_shape = tuple(np.array(wcs_shape) + 1)
+            pixel_inputs = np.meshgrid(*[np.arange(i) - 0.5 for i in wcs_shape], indexing='ij')
+        else:
+            pixel_inputs = np.meshgrid(*[np.arange(i) for i in wcs_shape], indexing='ij')
+
+        # Get world coords for all axes and all pixels.
+        axes_coords = self.wcs.pixel_to_world(*pixel_inputs)
+
+        # Reduce duplication across independent dimensions for each coord
+        # and transpose to make dimensions mimic numpy array order rather than WCS order.
+        for i, axis_coord in enumerate(axes_coords):
+            slices = np.array([slice(None)] * self.wcs.world_n_dim)
+            slices[np.invert(self.wcs.axis_correlation_matrix[i])] = 0
+            axis_coords[i] = axis_coord[tuple(slices)].T
+
+        world_axis_physical_types = self.wcs.world_axis_physical_types
+        # If user has supplied axes, extract only the
+        # world coords that correspond to those axes.
+        if axes:
+            # Convert input axes to WCS world axis indices.
+            if not isinstance(axes, tuple):
+                axes = (axes,)
+            world_indices = []
```

Comment:
why not make this a set up front?
```suggestion
            world_indices = set()
```

Reply:
Hadn't thought of that!